Create Safe Core SDK guide (#163)
safe-global/safe-docs: learn/safe-tools/safe-protocol-kit-safe-core-sdk/README.md

# Safe Protocol Kit (Safe Core SDK)

> The Safe Protocol Kit uses the [Safe Core S](https://github.com/safe-global/safe-core-sdk/tree/main/packages/safe-core-sdk)DK to interact with [Safe contracts](https://github.com/safe-global/safe-contracts) using a Typescript interface. This SDK can be used to create Safe accounts, and propose and execute transactions.

Reviewer: Link is not on complete SDK word

Author: Thanks! Fixed!

### Send ETH to The Safe

We will send some ETH to this Safe (treasury). Owner 1 will deposit 0.1 Goerli ETH to this Safe from our personal account following the [instructions in Quickstart](https://docs.gnosis-safe.io/learn/quickstart).

```tsx
const treasury = safeSdk.getAddress()

const treasuryAmount = ethers.utils.parseUnits("0.1", 'ether').toHexString()

const params = [{
  to: treasury,
  value: treasuryAmount
}]

const tx = await owner1Signer.sendTransaction(transactionParameters);

console.log("Fundraising.")
console.log(`Deposit Transaction: https://goerli.etherscan.io/tx/${tx.hash}`))
```

## Making a Transaction from a Safe

Owner 1 will sign and propose a transaction to send 0.05 ETH out of the Safe. Then, owner 2 will add their own proposal and execute the transaction since it meets the 2 of 3 thresholds.

At a high level, making a transaction from the Safe requires the following steps:

### Overview

1. Owner 1 creates a transaction
    1. Define the amount, destination, and any additional data to include in the transaction

>     2. Deterministically generate a hash

Reviewer 2: ```suggestion
```

2. Owner 1 proposes a transaction
    1. Performs an off-chain signature of the transaction before proposing
    2. Submits it to the Safe Service

>     3. Adds the signature to get 1 of 2 threshold

Reviewer 2: ```suggestion
```

3. Owner 2  gets pending transactions from the Safe service
4. Owner 2  confirms the transaction

>     1. Perform off-chain signature

Reviewer 2: ```suggestion
    1. Performs an off-chain signature
```

    2. Submits to service

> 5. Anyone executes the transaction (Owner 1 in this example)
>     1. On-chain transaction
>     2. The only step that is submitted to the blockchain
>     3. Gas is paid

Reviewer 2: ```suggestion
5. Anyone can execute the transaction (Owner 1 in this example)
```
As these are comments rather than steps the user has to follow, I think it's better to add a short description about them (I would say 3 is irrelevant) in the corresponding section

  senderAddress,
  senderSignature: senderSignature.data,
  origin
});
```

### Confirm the transaction: First Confirmation

```tsx
// transaction: SafeMultisigTransactionResponse

const hash = transaction.safeTxHash
let signature = await safeSdkOwner1.signTransactionHash(hash)
await safeService.confirmTransaction(hash, signature.data)
```

### Get Pending Transactions

Owner 2 needs a different Safe object. However, you don’t need to create it with the safe factory; you can create it with the `create` method of the Safe object. The Safe smart contract is already live on the blockchain so you can just pass the treasury address used when you created the Safe.

```tsx
const ethAdapterOwner2 = new EthersAdapter({
  ethers,
  signerOrProvider: owner2Signer
})

const safeSdkOwner2 = await Safe.create({ ethAdapter: ethAdapterOwner2,
                                          safeAddress: treasury });

const pendingTxs = await safeService.getPendingTransactions(safeAddress);

```

### Confirm the Transaction: Second Confirmation

```tsx
// Assuming that the first pending transaction is the one proposed by owner 1
const transaction = pendingTxs[0];
const safeTxHash = transaction.safeTxHash

> let signature = await safeSdkOwner2.signTransactionHash(hash)

Reviewer 2: ```suggestion
const signature = await safeSdkOwner2.signTransactionHash(hash)
```
